pkg/coalescer/coalescer_test.go

```diff
@@ -63,6 +66,7 @@ func TestCoalesceEvents(t *testing.T) {
 	wg.Wait()
 	// wait at least 1 debounce cycle for the final emission of events
 	time.Sleep(debounceDuration)
+	stop <- struct{}{}
```

Comment:
this is synchronous, because your channel is not buffered. Thus, we will block here until the goroutine on L54 reads this value.